### PR TITLE
removing noReply

### PR DESCRIPTION
**Pull request by franciscolopezsancho**
References #718

### CODE REVIEW

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/effect/SecondaryEffectImpl.scala [hunk 1]
```diff
@@ -43,21 +40,20 @@ sealed trait SecondaryEffectImpl {
...
     }
   }
 }
 
 case object NoSecondaryEffectImpl extends SecondaryEffectImpl {
   override def sideEffects: Vector[SideEffect] = Vector.empty
 
-  override def addSideEffects(sideEffects: Iterable[SideEffect]): SecondaryEffectImpl =
-    NoReply(sideEffects.toVector)
+  override def addSideEffects(newSideEffects: Iterable[SideEffect]): SecondaryEffectImpl =
```
Comment (franciscolopezsancho) [inline sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/effect/SecondaryEffectImpl.scala:52]: I'm not sure what to do with NoSecondaryEffectImpl itself. Is the default value for `_secondaryEffect` in `EventSourceEntityEffectImpl` for example

` private var _secondaryEffect: SecondaryEffectImpl = NoSecondaryEffectImpl`

**Comment by franciscolopezsancho** [comment]
I need to update the [akkaserverless-framework](https://github.com/lightbend/akkaserverless-framework)
For example [this](https://github.com/lightbend/akkaserverless-framework/blob/f420cd506649b7d312251231a3b2da74264edfc2/tck/src/main/scala/com/akkaserverless/tck/ActionTCK.scala#L114) test now wouldn't pass. I think I just need to delete those sort of tests . Those test testing `noReply` plus, of course, noReply method itself, such as [this](https://github.com/lightbend/akkaserverless-framework/blob/f420cd506649b7d312251231a3b2da74264edfc2/testkit/src/main/scala/com/akkaserverless/testkit/action/ActionMessages.scala#L53)

Would you agree?

**Comment by johanandren** [comment]
I agree, and then that will also require a release of the proxy and update of the version here in the SDK before this PR can be merged (possibly some change in the JS SDK as well, not sure)

**Comment by franciscolopezsancho** [comment]
local tests using https://github.com/lightbend/akkaserverless-framework/pull/1205/files pass. 

Pending PR above gets approved and with it, the new version of the framework is published. So, the CI tests running here --pointing to that new akkaserverless-framework image-- can succeed.

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/effect/SecondaryEffectImpl.scala (hunk 1 above)
Comment (johanandren) [reply sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/effect/SecondaryEffectImpl.scala:51]: I think it'd make more sense to make that an option/optional in the internal effect builder impls now that it is not possible to attach side effects to a no-reply, but maybe I'm missing something?

**Comment by franciscolopezsancho** [comment]
> I agree, and then that will also require a release of the proxy and update of the version here in the SDK before this PR can be merged (possibly some change in the JS SDK as well, not sure)

Yep, I just found out that I have to update `javascript-sdk` for integration tests on `framework` to pass.

**Comment by franciscolopezsancho** [comment]
javascript and framework ready
https://github.com/lightbend/akkaserverless-javascript-sdk/pull/255
https://github.com/lightbend/akkaserverless-framework/pull/1205

**Comment by franciscolopezsancho** [comment]
I'll try again against the new kalix-proxy

**Pull request closed without merging**